src/explorerCommands/rename.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as path from "path";
+import { Uri, window, workspace, WorkspaceEdit } from "vscode";
+import { DataNode } from "../views/dataNode";
+import { ExplorerNode } from "../views/explorerNode";
+import { isMutable } from "./utils";
+
+export async function renameFile(node: DataNode, selectedNode: ExplorerNode): Promise<void> {
+    // if command not invoked by context menu, use selected node in explorer
+    if (!node) {
+        node = selectedNode as DataNode;
+        if (!isMutable(node)) {
+            return;
+        }
+    }
+
+    const newName: string | undefined = await window.showInputBox({
+        placeHolder: "Input new file name",
+        ignoreFocusOut: true,
+    });
+
+    if (!newName) {
+        return;
+    }
+
+    const oldFsPath = Uri.parse(node.uri).fsPath;
+    const renamedFilePath = getRenamedFilePath(oldFsPath, newName);
+
+    const workspaceEdit: WorkspaceEdit = new WorkspaceEdit();
+    workspaceEdit.renameFile(Uri.file(oldFsPath), Uri.file(renamedFilePath));
+    workspace.applyEdit(workspaceEdit);
+}
+
+function getRenamedFilePath(oldUri: string, newName: string): string {
+    // preserve default file extension if not provided
+    if (!path.extname(newName)) {
```

Comment:
what will happen if the rename happens on a file which does not have a file extension name originally? For example, rename file `A` -> `B`

Reply:
It will behave normally, since `path.extname` will return an empty string if no extension name provided.

![image](https://user-images.githubusercontent.com/30518415/96541116-e4249680-12d1-11eb-8c67-ce59844036e9.png)